src/value.cc

```diff
@@ -567,29 +564,24 @@ bool Value::getVec3(double &x, double &y, double &z, double defaultval) const
     if (v.size() != 3) return false;
   }
 
-  return (v[0]->getDouble(x) && v[1]->getDouble(y) && v[2]->getDouble(z));
+  return (v[0].getDouble(x) && v[1].getDouble(y) && v[2].getDouble(z));
 }
 
-RangeType Value::toRange() const
+const RangeType& Value::toRange() const
 {
+  static const RangeType empty(0,0,0);
   const RangeType *val = boost::get<RangeType>(&this->value);
   if (val) {
     return *val;
   }
-  else return RangeType(0,0,0);
+  else return empty;
 }
 
-Value &Value::operator=(const Value &v)
-{
-  if (this != &v) {
-    this->value = v.value;
-  }
-  return *this;
-}
 
 class equals_visitor : public boost::static_visitor<bool>
 {
 public:
+	typedef bool result_type;
```

Comment:
This was something I noticed that boost technically specifies as required for static visitors
[reference](https://www.boost.org/doc/libs/1_64_0/doc/html/variant/reference.html#variant.concepts.static-visitor): see second point
> The requirements on a static visitor of a type T are as follows:
> 
> - Must allow invocation as a function by overloading operator(), unambiguously accepting any value of type T.
> - Must expose inner type result_type. C++14 compatible compilers could detect result_type automatically, but will stick to result_type if it is defined. (See boost::visitor_ptr for a solution to using functions as visitors.)
> - If result_type is not void, then each operation of the function object must return a value implicitly convertible to result_type.

I added this to see if it would help with the Trusty compile issues, but anyways as I found out, its due to the boost version not supporting move-only types as return values for visitors, so adding these definitions wasn't a solution to the problem.  Its been working fine without it so far, so I'm fine either removing it again or, or maybe just documenting in comments that boost technically requires it (but doesn't really need it i guess).

Reply:
I ended up removing these in https://github.com/openscad/openscad/pull/3102  since we're officially on C++14, and boost shouldn't need them.